src/Aspire.Dashboard/Components/Pages/ComponentTelemetryContext.cs

```diff
@@ -15,9 +16,10 @@ public class ComponentTelemetryContext(string componentType) : IDisposable
 
     private DashboardTelemetryService TelemetryService => _telemetryService ?? throw new ArgumentNullException(nameof(_telemetryService), "InitializeAsync has not been called");
 
-    public async Task InitializeAsync(DashboardTelemetryService telemetryService)
+    public async Task InitializeAsync(DashboardTelemetryService telemetryService, IJSRuntime js)
     {
         Properties[TelemetryPropertyKeys.DashboardComponentId] = new AspireTelemetryProperty(componentType);
+        Properties[TelemetryPropertyKeys.UserAgent] = new AspireTelemetryProperty(await telemetryService.GetUserAgentAsync(js));
```

Comment:
I haven't got up to `GetUserAgentAsync` yet, but just want to make sure that the value is cached. Add a comment here that GetUserAgentAsync is cached.

Reply:
When the dashboard is first visited some info is sent from the client, such as the browser timezone. I think the user agent could be sent at the same time. That batches together client interop calls and improves initial load performance.